actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -171,6 +209,26 @@ def url_from(location)
       location if location && _url_host_allowed?(location)
     end
 
+    # Returns the value of the +onward+ query param if it is an internal URL or
+    # path. Otherwise returns +nil+.
+    #
+    # This method is also exposed as a view helper.
+    #
+    # ==== Examples
+    #
+    #   # Responding to `GET /sign_in?onward=%2Fposts%2F1%2Fedit`:
+    #   onward_url # => "/posts/1/edit"
+    #
+    #   # Responding to `GET /sign_in`:
+    #   onward_url # => nil
+    #
+    #   # Responding to `GET /sign_in?onward=http%3A%2F%2Fevil.com`:
+    #   onward_url # => nil
+    #
+    def onward_url
```

Comment:
We could enhance `onward_url` to support an `allow_other_host:` kwarg, but keeping it simple for now.